OneSchedule.Tests/OnesheduleStateTests.cs

```diff
@@ -0,0 +1,225 @@
+            var dto = new DtoDomain() { ChatId = chatId };
+
+            _stateContextMoq.Setup(o => o.ContextEntity).Returns(new ContextEntity() { Event = new EventEntity() { Id = "55566" } });
+            _botMoq.Setup(o => o.SendTextMessageAsync(chatId, It.IsAny<string>(), ParseMode.Default, null, false, false, 0, false, It.IsAny<ReplyKeyboardMarkup>(), default));
+
+            await state.HandleAsync(_stateContextMoq.Object, dto);
+
+            _botMoq.Verify(o => o.SendTextMessageAsync(chatId, It.IsAny<string>(), ParseMode.Default, null, false, false, 0, false, It.IsAny<ReplyKeyboardMarkup>(), default), Times.Once);
+            Assert.AreEqual(null, _stateContextMoq.Object.ContextEntity.LastState);
+        }
+
+        [Test]
+        public async Task AskShowEventListState_IsCompleted()
+        {
+            var repository = new Mock<IRepository<EventEntity>>();
+            repository.Setup(o => o.FindAsync(It.IsAny<Expression<Func<EventEntity, bool>>>())).Returns(Task.Run(() => new List<EventEntity>()));
+            _botMoq = new Mock<ITelegramBotClient>();
+            var chatId = "123";
+            var state = new AskShowEventListState(_botMoq.Object, repository.Object);
+            var dto = new DtoDomain() { ChatId = chatId };
+
+            _stateContextMoq.Setup(o => o.ContextEntity).Returns(new ContextEntity() { Event = new EventEntity() { Id = "55566" } });
+            _botMoq.Setup(o => o.SendTextMessageAsync(chatId, It.IsAny<string>(), ParseMode.Default, null, false, false, 0, false, It.IsAny<ReplyKeyboardMarkup>(), default));
+
+            await state.HandleAsync(_stateContextMoq.Object, dto);
+
+            _botMoq.Verify(o => o.SendTextMessageAsync(chatId, It.IsAny<string>(), ParseMode.Default, null, false, false, 0, false, It.IsAny<ReplyKeyboardMarkup>(), default), Times.Once);
+            Assert.AreEqual(null, _stateContextMoq.Object.ContextEntity.LastState);
+        }
+
+        [Test]
+        public async Task GetDescriptionState_IsCompleted()
+        {
+            var chatId = "123";
+            var dto = new DtoDomain() { ChatId = chatId };
+            var state = new GetDescriptionState();
+
+            _stateContextMoq.Setup(o => o.ContextEntity).Returns(new ContextEntity(){Event = new EventEntity()});
+            
+            await state.HandleAsync(_stateContextMoq.Object, dto);
+
+            Assert.AreEqual("GetDescription", _stateContextMoq.Object.ContextEntity.LastState);
+        }
+        
+        [Test]
+        public void GetBeginDateState_WrongDateFormat()
+        {
+            var chatId = "123";
+            var dto = new DtoDomain() { ChatId = chatId,MessageText = "tnn"};
+            var state = new GetBeginDateState();
+
+            _stateContextMoq.Setup(o => o.ContextEntity).Returns(new ContextEntity(){Event = new EventEntity()});
+
+            Assert.Catch<BotAppInternalException>(() => state.HandleAsync(_stateContextMoq.Object, dto));
```

Comment:
If you want to check that a particular exception was thrown you need to use Throws method.
Here is a references to the docs where you can read what the difference between them
https://docs.nunit.org/articles/nunit/writing-tests/assertions/classic-assertions/Assert.Throws.html
https://docs.nunit.org/articles/nunit/writing-tests/assertions/classic-assertions/Assert.Catch.html